test/unit/util/locking/RedisResourceLocker.test.ts

```diff
@@ -0,0 +1,169 @@
+// eslint-disable-next-line import/default
+import redis from 'redis';
+import type { RedisClient } from 'redis';
+import { Lock } from 'redlock';
+import { InternalServerError } from '../../../../src/util/errors/InternalServerError';
+import { RedisResourceLocker } from '../../../../src/util/locking/RedisResourceLocker';
+
```

Comment:
You need to have a `jest.mock('redis')` here otherwise you are overwriting the functions of the actual redis library below, which could potentially impact other tests that do require the actual redis library, depending on the order the tests are being run.

Reply:
My solution is based on: https://github.com/facebook/jest/issues/8983

I thought this overwrite would only effect this testsuite. Would some `clearAllMocks()` function in the AfterAll() fix this?

Reply:
Ah that is an unfortunate bug. Definitely link to that issue in unit tests to mention why it is done as it is. 

If the integration tests work that's fine, but at some point there were unit tests that mocked the `fs` function in a similar fashion, which caused the integrations tests that used the file system to start failing.

I don't think `clearAllMocks` would solve that issue since you are overwriting a field of `redis` here.

Make sure that at least 1 integration test would fail if it would be using the mocked version instead of the actual redis import so we can quickly detect if this is a problem, or becomes one in the future.